staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -393,8 +393,10 @@ type CustomResourceSubresourceScale struct {
 	StatusReplicasPath string
 	// LabelSelectorPath defines the JSON path inside of a CustomResource that corresponds to Scale.Status.Selector.
 	// Only JSON paths without the array notation are allowed.
-	// Must be a JSON Path under .status.
 	// Must be set to work with HPA.
+	// The field pointed by this JSON path must be a string field (not a complex selector struct) and contain
+	// a serialized, version-agnostic label query over objects that should match the replicas count.
```

Comment:
why version agnostic?

Reply:
the "version" here is not CR version but Spec.Selector version (schema). Spec.Selector schema varies across different objects and API versions, and ScaleStatus.Selector is intended to abstract away the different forms of Spec.Selector

I copied the word from https://github.com/kubernetes/kubernetes/issues/66688#issuecomment-425996199:

> where in your Status do you report to us the serialized, version-agnostic form of your Spec.Selector

> Spec.SerializedVersionAgnosticSelector which is in the format required for ScaleStatus.Selector

I can phrase it as "spec-selector-schema-agnostic". Other suggestions are welcome

Reply:
maybe something like 

> The field pointed by this JSON path must be a string field (not a complex selector struct) which contains a serialized label selector in string form.